core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -120,6 +121,9 @@ public static IncludeExclude parseInclude(XContentParser parser, QueryParseConte
                             "Unknown parameter in Include/Exclude clause: " + currentFieldName);
                 }
             }
+            if (pattern != null) {
+                return new IncludeExclude(pattern, null);
```

Comment:
Maybe throw an exception here if partition criteria supplied?
Partitioning not supported when using regex-based filtering.